base_layer/core/src/proof_of_work/diff_adj_manager/diff_adj_storage.rs

```diff
@@ -314,43 +328,22 @@ impl DiffAdjStorage {
     ) -> Result<(), DiffAdjManagerError>
     {
         self.reset();
-        let mut monero_diff_list = Vec::<(EpochTime, Difficulty)>::with_capacity(self.difficulty_block_window as usize);
-        let mut blake_diff_list = Vec::<(EpochTime, Difficulty)>::with_capacity(self.difficulty_block_window as usize);
-        for height in (0..=height_of_longest_chain).rev() {
+        for height in 0..=height_of_longest_chain {
```

Comment:
This should be in reverse as you only want to use the latest samples as the loop will break and you won't use the whole range.

Reply:
I took out another reverse further down as well, so the effect should be the same

Reply:
It is not entirely the same as you only need to query the top +-300 headers to estimate the new target difficulty and not the entire chain.

Reply:
With this method you need the entire history, and you always have to build up from 0

Reply:
Correct, because we do not store the target difficulty anywhere. As a secondary PR we can store it somewhere and retrieve it. I don't think this impacts on the correctness of this PR